### PR TITLE
Support request method and headers in fetch()

### LINKED ISSUE #1133: POST requests with fetch aren't working
Where in this case, "not working" means "isn't sending headers to the server", as near as I can tell.

Reproduction: https://gist.github.com/TehShrike/c8afda9c15b7061e4a57b7d9d7f9777e

This is blocking #1123

### PR DESCRIPTION
**Pull request by ry**
Adds a general HttpHeader flatbuffer message for serializing requests
and responses.

Fixes #1133

### CODE REVIEW

File: js/headers.ts [hunk 1]
```diff
@@ -79,7 +79,8 @@ class HeadersBase {
...
 // tslint:disable-next-line:variable-name
-export const Headers = DomIterableMixin<string, string, typeof HeadersBase>(
-  HeadersBase,
-  headerMap
-);
+export class Headers extends DomIterableMixin<
+  string,
+  string,
+  typeof HeadersBase
+>(HeadersBase, headerMap) {}
```
Comment (ry) [inline js/headers.ts]: @kitsonk Is this ok with you? I was getting type errors that were resolved when I did this.

File: src/msg.fbs [hunk 2]
```diff
@@ -194,18 +194,26 @@ table KeyValue {
   value: string;
 }
 
-table Fetch {
-  id: uint;
+// Note this represents The WHOLE header of an http message, not just the key
```
Comment (piscisaureus) [inline src/msg.fbs]: What's the advantage of doing this vs separating the status/method out of this table?

Comment (ry) [reply src/msg.fbs]: The idea is to reuse this message for server-side messages too. If url/method/status is put into the table, there will be duplicate code to extract those in the server/client.

**Comment by ry** [comment]
@piscisaureus test added.

File: src/msg_util.rs [hunk 3]
```diff
@@ -15,6 +26,99 @@ pub fn serialize_key_value<'bldr>(
...
+  *r.uri_mut() = uri;
+
+  if let Some(method) = header_msg.method() {
+    let method = Method::from_str(method).unwrap();
+    *r.method_mut() = method;
+  }
+
+  if let Some(fields) = header_msg.fields() {
+    let headers = r.headers_mut();
+    for i in 0..fields.len() {
```
[the remark this answers is not in the document]
Comment (ry) [reply src/msg_util.rs]: There's no iter for `flatbuffers::Vector`

File: js/headers.ts (hunk 1 above)
Comment (kitsonk) [reply js/headers.ts]: Wow, I honestly didn't realise that worked, so no, no problem at all. 👍

File: js/fetch_test.ts [hunk 4]
```diff
@@ -46,3 +46,34 @@ testPerm({ net: true }, async function responseClone() {
...
+    headers: [["Hello", "World"], ["Foo", "Bar"]]
+  });
+  listener.close();
+  assertEqual(response.status, 404);
+  assertEqual(response.headers.get("Content-Length"), "2");
+
+  const actual = new TextDecoder().decode(buf.bytes());
+  const expected = [
+    "POST /blah HTTP/1.1\r\n",
+    "hello: World\r\n",
```
Comment (piscisaureus) [inline js/fetch_test.ts]: Does it really lower-case the headers? :/

**Pull request merged**